app/utils/gallery/vectors.ts

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import Animated, {useSharedValue} from 'react-native-reanimated';
+
+type SharedValueType = number;
+
+type Callback = () => unknown;
+
+type VectorType = Vector<any> | SharedVector<any>;
+
+type VectorList = Array<VectorType | SharedValueType>;
+
+type Operation = 'divide' | 'add' | 'sub' | 'multiply';
+
+type VectorProp = 'x' | 'y';
+
+export type Vector<T extends SharedValueType> = {
+    x: T;
+    y: T;
+};
+
+export type SharedVector<T extends SharedValueType> = {
+    x: Animated.SharedValue<T>;
+    y: Animated.SharedValue<T>;
+};
+
+const isVector = (value: any): value is Vector<any> => {
+    'worklet';
+
+    return typeof value.x !== 'undefined' && typeof value.y !== 'undefined';
```

Comment:
Any strong reason to do `typeof x !== 'undefined'` instead of `x !== undefined`?

Reply:
no strong reason at all.. I can change it but the effect is the same